x-pack/legacy/plugins/siem/public/components/auto_sizer/index.tsx

```diff
@@ -5,7 +5,7 @@
  */
```

Comment:
To get my head around these changes I had to write it too and came up with that. it is mostly about naming and not using the `isMeasiring` since it won't detect the unMount since the value is already to true. let me know what do you think?
```
export const AutoSizer = React.memo<AutoSizerProps>(
  ({ bounds = false, children, content = true, detectAnyWindowResize, onResize }) => {
    const element = useRef<HTMLElement | null>(null);
    const resizeObserver = useRef<ResizeObserver | null>(null);
    const windowWidth = useRef<number>(-1);
    const [boundsMeasurement, setBoundsMeasurement] = useState<Measurement>({
      height: 0,
      width: 0,
    });
    const [contentMeasurement, setContentMeasurement] = useState<Measurement>({
      height: 0,
      width: 0,
    });
    const [windowMeasurement, setWindowMeasurement] = useState<Measurement>({
      height: 0,
      width: 0,
    });

    const storeRef = (htmlElement: HTMLElement | null) => {
      if (htmlElement && resizeObserver.current) {
        resizeObserver.current.observe(htmlElement);
      }
      element.current = htmlElement;
    };

    const measure = (entry: ResizeObserverEntry | null) => {
      if (!element.current) {
        return;
      }
      const boundsRect = bounds ? element.current.getBoundingClientRect() : null;

      const boundsMeasurementNow: Measurement = boundsRect
        ? {
            height: element.current.getBoundingClientRect().height,
            width: element.current.getBoundingClientRect().width,
          }
        : boundsMeasurement;

      const windowMeasurementNow: Measurement = {
        width: window.innerWidth,
        height: window.innerHeight,
      };

      if (
        detectAnyWindowResize &&
        boundsMeasurementNow &&
        boundsMeasurementNow.width &&
        windowWidth.current !== -1 &&
        windowWidth.current > window.innerWidth
      ) {
        const gap = windowWidth.current - window.innerWidth;
        boundsMeasurement.width = boundsMeasurementNow.width - gap;
      }
      windowWidth.current = window.innerWidth;
      const contentRect = content && entry ? entry.contentRect : null;
      const contentMeasurementNow: Measurement =
        contentRect && entry
          ? {
              height: entry.contentRect.height,
              width: entry.contentRect.width,
            }
          : contentMeasurement;

      if (
        isEqual(boundsMeasurementNow, boundsMeasurement) &&
        isEqual(contentMeasurementNow, contentMeasurement) &&
        isEqual(windowMeasurementNow, windowMeasurement)
      ) {
        return;
      }

      requestAnimationFrame(() => {
        if (!resizeObserver) {
          return;
        }

        setBoundsMeasurement(boundsMeasurementNow);
        setContentMeasurement(contentMeasurementNow);
        setWindowMeasurement(windowMeasurementNow);
        if (onResize) {
          onResize({
            bounds: boundsMeasurementNow,
            content: contentMeasurementNow,
            windowMeasurement: windowMeasurementNow,
          });
        }
      });
    };

    useEffect(() => {
      if (detectAnyWindowResize) {
        window.addEventListener('resize', () => measure(null));
      }

      resizeObserver.current = new ResizeObserver(entries => {
        entries.forEach(entry => {
          if (entry.target === element.current) {
            measure(entry);
          }
        });
      });

      return () => {
        if (resizeObserver && resizeObserver.current) {
          resizeObserver.current.disconnect();
          resizeObserver.current = null;
        }
        if (detectAnyWindowResize) {
          window.removeEventListener('resize', () => measure(null));
        }
      };
    }, []);

    return children({
      bounds: boundsMeasurement,
      content: contentMeasurement,
      windowMeasurement,
      measureRef: storeRef,
    });
  }
);
```

Reply:
implemented 👍